python/IECore/Enum.py

```diff
@@ -82,7 +82,9 @@ def __hash__( self ) :
 
 		def __eq__( self, other ):
 
-			assert( type( self ) is type( other ) )
+			if type( self ) is not type( other ):
+				return False
```

Comment:
Great! Sorry, I should have realised this would have been better in the original review. This is also equivalent to the behaviour of Python 3's enum class.